src/js/timers.js

```diff
@@ -29,7 +29,13 @@ function Timeout(after) {
 native.prototype.handleTimeout = function() {
   var timeout = this.timeoutObj; // 'this' is native object
   if (timeout && timeout.callback) {
-    timeout.callback();
+    try {
+      timeout.callback();
+    } catch (e) {
+      timeout.unref();
```

Comment:
Don't we need `if (!timeout.isrepeat)` here as well? An exception aborts a periodic timer?

Reply:
Yes, an exception must abort a periodic timer too.